### PR TITLE
Bundle Minify Update

### LINKED ISSUE #1786: Update Bundling and Minification
- [ ] Update Gulp & Grunt portions
- [ ]  Add Bundler Minifier : https://visualstudiogallery.msdn.microsoft.com/9ec27da7-e24b-4d56-8064-fd7e88ac1c40 and https://github.com/madskristensen/BundlerMinifier
- [ ] Update to match current templates
- [ ] Break in to Intro, 3 sub parts Bundler Minifier, Gulp, Grunt
- [ ] Covers #1751 

cc @danroth27

### PR DESCRIPTION
**Pull request by spboyer**
Fixes #1786 

@madskristensen @danroth27

### CODE REVIEW

**Comment by spboyer** [comment]
@danroth27 - can you review this so we can get it pushed out asap.

**Comment by danroth27** [comment]
@spboyer I suspect there shouldn't be 300+ files in this update

**Comment by spboyer** [comment]
@danroth27 - This was PR against csproj before the merge to live, the conflict was probably due to that issue. Should I cancel and re-PR against Live?

File: aspnetcore/client-side/bundling-and-minification.md [hunk 1]
```diff
@@ -14,46 +14,21 @@ uid: client-side/bundling-and-minification
...
+In ASP.NET Core apps, there are options for bundling and minifying client-side resources. The core templates for MVC provide an out of the box solution using a configuration file and BuildBundlerMinifier NuGet package. Third party tools, such as [Gulp](using-gulp.md) and [Grunt](using-grunt.md) are also avaiable to accomplish the same tasks should your processes require additional workflow or complexities. By using design-time bundling and minification, the minified files are created prior to the application’s deployment. Bundling and minifying before deployment provides the advantage of reduced server load. However, it’s important to recognize that design-time bundling and minification increases build complexity and only works with static files.
```
Comment (danroth27) [inline aspnetcore/client-side/bundling-and-minification.md]: s/avaiable/available

Comment (danroth27) [inline aspnetcore/client-side/bundling-and-minification.md]: there are options -> there are multiple

File: aspnetcore/client-side/bundling-and-minification.md [hunk 2]
```diff
@@ -86,36 +61,120 @@ imageTagAndImageID | t
...
+1>
+1>Bundler: Begin processing bundleconfig.json
+1>Bundler: Done processing bundleconfig.json
+1>BuildBundlerMinifierExample -> C:\BuildBundlerMinifierExample\bin\Debug\netcoreapp1.1\BuildBundlerMinifierExample.dll
+========== Build: 1 succeeded or up-to-date, 0 failed, 0 skipped ==========
+```
+
+### Visual Studio Code or Command Line
+
+Visual Studio and the extension drives the process and using GUI gestures, however the same capabilities are available with the `dotnet` CLI and BuildBundlerMinifier NuGet Package.
```
Comment (danroth27) [inline aspnetcore/client-side/bundling-and-minification.md]: > Visual Studio and the extension drives the process and using GUI gestures

Visual Studio and the extension drive the bundling and minification process using GUI gestures

File: aspnetcore/client-side/bundling-and-minification.md [hunk 3]
```diff
@@ -86,36 +61,120 @@ imageTagAndImageID | t
...
+
+  Bundler: Begin processing bundleconfig.json
+     Minified wwwroot/css/site.min.css
+  Bundler: Done processing bundleconfig.json
+  BuildBundlerMinifierExample -> /BuildBundlerMinifierExample/bin/Debug/netcoreapp1.0/BuildBundlerMinifierExample.dll
+```
+
+## Adding files
+
+In this example, an additional css file is added called `custom.css` and configured for bundling and minification with `site.css`, resulting in a single `site.min.css`.
```
Comment (danroth27) [inline aspnetcore/client-side/bundling-and-minification.md]: Use a consistent casing for CSS

File: aspnetcore/client-side/bundling-and-minification.md [hunk 4]
```diff
@@ -131,6 +190,29 @@ This environment tag will render the bundled and minified CSS files only when ru
 
 [!code-html[Main](../client-side/bundling-and-minification/samples/WebApplication1/src/WebApplication1/Views/Shared/_Layout.cshtml?highlight=5&range=12-17)]
 
+## Consuming bundleconfig.json from Gulp
+
+If your app bundling and minification workflow requires additional processes such as image processing, cache busting or CDN assest processing etc., then you can convert the Bundle and Minify process to Gulp.
+
+> [!NOTE]
+> Conversion option only available in Visual Studio 2015 or 2017.
```
Comment (danroth27) [inline aspnetcore/client-side/bundling-and-minification.md]: 2015? We only support 2017.

File: aspnetcore/client-side/bundling-and-minification.md [hunk 5]
```diff
@@ -131,6 +190,29 @@ This environment tag will render the bundled and minified CSS files only when ru
...
+
+```xml
+<Target Name="MyPreCompileTarget" BeforeTargets="Build">
+    <Exec Command="gulp min" />
+</Target>
+```
+
 ## Additional resources
 
 * [Using Gulp](using-gulp.md)
```
Comment (danroth27) [inline aspnetcore/client-side/bundling-and-minification.md]: Remove the extra line breaks in the bulleted list

File: aspnetcore/client-side/bundling-and-minification.md (hunk 4 above)
Comment (spboyer) [inline aspnetcore/client-side/bundling-and-minification.md]: the option is available in 2015 as well and the extension in the marketplace states 2015, 2017.  However, changing for overall support messaging.

File: aspnetcore/client-side/bundling-and-minification.md (hunk 5 above)
Comment (spboyer) [inline aspnetcore/client-side/bundling-and-minification.md]: done

File: aspnetcore/client-side/bundling-and-minification.md (hunk 3 above)
Comment (spboyer) [inline aspnetcore/client-side/bundling-and-minification.md]: fixed

File: aspnetcore/client-side/bundling-and-minification.md (hunk 2 above)
Comment (spboyer) [inline aspnetcore/client-side/bundling-and-minification.md]: fixed.

File: aspnetcore/client-side/bundling-and-minification.md (hunk 1 above)
Comment (spboyer) [inline aspnetcore/client-side/bundling-and-minification.md]: fixed

Comment (spboyer) [inline aspnetcore/client-side/bundling-and-minification.md]: fixed

**Comment by spboyer** [comment]
Closing this PR and creating new against master.

**Pull request closed without merging**